### DISCUSSION TITLE
Load workflow from json file

### DISCUSSION THREAD

**Discussion by thiagoteles** [elsa-workflows/elsa-core#506] [Q&A] [answered]
I'm trying to create a integration tests using ef in memory database provider but we are using the dashboard to create the workflows. How can I export json workflow representation from dashboard and import in my test initialize?

**Comment by sfmskywalker** [comment]
Happy to see you found it on gitter. Thanks for posting it here, I will convert this to a Q&A discussion for future reference.

**Comment by sfmskywalker** [comment]
@thiagoteles Perhaps you would consider changing your question back to its original text, and then add your answer as a separate comment & mark it as The Answer?

**Reply by thiagoteles** [reply]
Sure!

**Comment by thiagoteles** [comment, marked as answer]
Edit: I found a topic in gitter. 

I`ll post here, maybe I can help someone.

1. Create a class: 

```
public class JsonWorkflowProvider : IWorkflowProvider
{
    private readonly IWorkflowSerializer _serializer;
    private readonly IConfiguration _configuration;

    public JsonWorkflowProvider(IWorkflowSerializer serializer, IConfiguration configuration)
    {
        _serializer = serializer;
        _configuration = configuration;
    }
    
    public async Task<IEnumerable<WorkflowDefinitionVersion>> GetWorkflowDefinitionsAsync(CancellationToken cancellationToken)
    {
        var workflowsDirectoryPath = _configuration["WorkflowPath"];
        var workflowFiles = Directory.GetFiles(workflowsDirectoryPath, "*.json");
        var workflowDefinitionTasks =  workflowFiles.Select(LoadWorkflowDefinition).ToList();
        await Task.WhenAll(workflowDefinitionTasks);
        return workflowDefinitionTasks.Select(x => x.Result);
    }

    private async Task<WorkflowDefinitionVersion> LoadWorkflowDefinition(string path)
    {
        var jsonText = await File.ReadAllTextAsync(path);
        return _serializer.Deserialize<WorkflowDefinitionVersion>(jsonText, JsonTokenFormatter.FormatName);
    }
}
```

2. Then simply register it as a singleton with DI
```
services.AddSingleton<IWorkflowProvider, JsonWorkflowProvider>();
```

**Reply by chandramohank** [reply]
@sfmskywalker  any examples available to load the workflow from json using Elsa 2.0

**Reply by jdevillard** [reply]
Hello, 

you could use the Serialization Sample.

https://github.com/elsa-workflows/elsa-core/tree/master/src/samples/console/Elsa.Samples.Serialization


The JSON is the serialization of the WorkflowDefinition.

At the end of the [program.cs](https://github.com/elsa-workflows/elsa-core/blob/967fbddf8b40b9fb67bb55362ee2496c07366e4f/src/samples/console/Elsa.Samples.Serialization/Program.cs#L62), you can deserialize and then launch you workflow

**Reply by sfmskywalker** [reply]
In addition, there's another hidden example that you just reminded me to document: https://github.com/elsa-workflows/elsa-core/blob/master/src/samples/aspnet/Elsa.Samples.SignalApi/Startup.cs#L29

That example shows how to setup a simple provider that reads JSON files from a directory. You use the designer to create & export JSON files and you just save them into the "workflows" folder.
